src/index.js

```diff
@@ -73,6 +75,49 @@ const renderFragment = async (path) => {
   };
 };
 
+/**
+ * Renders content from AEM UE pages
+ */
+const renderContent = async (path, params) => {
+  const { authorization } = params;
+
+  /**
+   * FIXME: Page extension to be handled dynamically. Revisit this implementation and identify a proper
+   * fix.
+   * */
+
+  const aemURL = `${aemConfig.aemEnv}${path}.html?wcmmode=disabled`;
+  // console.log(aemURL);
+  const url = new URL(aemURL);
+
+  const fetchHeaders = { 'cache-control': 'no-cache' };
+  if (authorization) {
+    fetchHeaders.authorization = authorization;
+  }
+
+  const resp = await fetch(url, { headers: fetchHeaders });
+
+  if (!resp.ok) {
+    return { error: { code: resp.status, message: resp.statusText } };
+  }
+
+  let contentType = resp.headers.get('content-type') || 'text/html';
+  [contentType] = contentType.split(';');
+
+  const respHeaders = {
+    'content-type': contentType,
+  };
+
+  if (isBinary(contentType)) {
```

Comment:
We should check if this is really necessary. Ideally we would not spool binaries through the converter. That means we rather check if the incoming request is .html or .json and if not fail with a [415 Unsupported Media Type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/415)